Fix #277: Private blank tabs now display users favorites

kylehickinson opened the pull request:
Also fixes an ugly animation issue when creating new tabs in private mode from the tab tray

## Pull Request Checklist

- [x] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`
- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [x] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

![image](https://user-images.githubusercontent.com/529104/46306618-c0734400-c582-11e8-80fa-234de8202f98.png)

## Notes for testing this patch

_None included_

jhreis reviewed (approved):
So much code deletion!

kylehickinson committed:
Fix #277: Private blank tabs now display users favorites

Also fixes an ugly animation issue when creating new tabs in private mode from the tab tray
